src/Chapter3.hs

```diff
@@ -971,6 +1071,27 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data WeekDay = Monday
+              | Tuesday
+              | Wednesday
+              | Thursday
+              | Friday
+              | Saturday
+              | Sunday
+              deriving (Show, Eq, Ord, Enum)
+
+isWeekend :: WeekDay -> Bool
+isWeekend Saturday = True
+isWeekend Sunday = True
+isWeekend _ = False
+
+nextDay :: WeekDay -> WeekDay
+nextDay Sunday = Monday --Not sure if enum will handle this already, including just to be safe
+nextDay day = succ day
+
+daysToParty :: WeekDay -> Int
+daysToParty day | fromEnum day < fromEnum Friday = fromEnum Friday - fromEnum day
+                | otherwise = 7 - (fromEnum day - fromEnum Friday)
```

Comment:
If you recall how `mod` works for negative numbers, you can implement this function in a shorter way :slightly_smiling_face: